rapid7/metasploit-framework
# Submission of New Exploit Module for Vinchin Backup & Recovery Command Injection

Hello Metasploit Team,

I am excited to submit a new exploit module for inclusion in the Metasploit Framework targeting a command injection vulnerability in Vinchin Backup & Recovery versions v5.0.*, v6.0.*, v6.7.*, and v7.0.*.

**Module Name**: exploit/linux/http/vinchin_backup_recovery_cmd_inject

### Module Description
This module exploits a command injection vulnerability in Vinchin Backup & Recovery software. It sends a specially crafted HTTP request to the server, resulting in arbitrary command execution under the context of the web server user. After extensive testing, I have determined that the `generic/shell_reverse_tcp` payload is uniquely suited to exploit this vulnerability effectively. Other payloads were tested but did not succeed, making this payload the only currently viable option for successful exploitation.

### Verification Steps:
1. Start `msfconsole`.
2. `use exploit/linux/http/vinchin_backup_recovery_cmd_inject`.
3. Set `RHOSTS`, `RPORT`, `SSL`, `TARGETURI`, `APIKEY`, `LHOST`, `LPORT`.
4. Run `check` to verify the target is vulnerable.
5. Run `exploit` to execute the payload on the target system.

### Scenarios
A comprehensive scenario is provided in the documentation, demonstrating the successful exploitation of a Vinchin Backup & Recovery system. This scenario also serves as a reference for the exploit's effectiveness when using the `generic/shell_reverse_tcp` payload. (Other payloads don't work in my lab)

I have closely followed the module development guidelines and included thorough documentation.

Please let me know if there are any required changes or additional tests that I could perform. I am open to feedback and willing to make any necessary revisions.

Thank you for considering my contribution to the Metasploit Framework.

Best regards,

Chocapikk
